shuffle/cli/src/shared.rs

```diff
@@ -407,44 +413,61 @@ pub struct NetworksConfig {
 }
 
 impl NetworksConfig {
-    pub fn default() -> Self {
+    pub fn default() -> Result<Self> {
         let mut network_map = BTreeMap::new();
-        network_map.insert("localhost".to_string(), Network::default());
-        NetworksConfig {
+        network_map.insert("localhost".to_string(), Network::localhost()?);
+        network_map.insert("trove".to_string(), Network::trove_testnet()?);
+        Ok(NetworksConfig {
             networks: network_map,
-        }
+        })
     }
 
-    pub fn url_for(&self, network_name: &str) -> Result<Url> {
-        let specified_network = self.networks.get(network_name).ok_or_else(|| {
+    pub fn get_network(&self, network_name: &str) -> Result<Network> {
+        Network::from(self.networks.get(network_name).ok_or_else(|| {
             anyhow!("Please add specified network to the ~/.shuffle/Networks.json")
-        })?;
-        Ok(Url::from_str(
-            format!(
-                "{}:{}",
-                specified_network.base, specified_network.dev_api_port
-            )
-            .as_str(),
-        )?)
+        })?)
     }
 }
 
 #[derive(Debug, Serialize, Deserialize, PartialEq)]
 pub struct Network {
     name: String,
-    base: String,
-    json_rpc_port: u16,
-    dev_api_port: u16,
+    json_rpc_url: Url,
+    dev_api_url: Url,
+    faucet_url: Option<Url>,
 }
 
 impl Network {
-    pub fn default() -> Self {
-        Network {
-            name: String::from(LOCALHOST_NETWORK_NAME),
-            base: String::from(LOCALHOST_NETWORK_BASE),
-            json_rpc_port: 8080,
-            dev_api_port: 8081,
-        }
+    pub fn localhost() -> Result<Self> {
+        Ok(Network {
+            name: String::from(LOCALHOST_NAME),
+            json_rpc_url: Url::from_str("http://127.0.0.1:8080")?,
+            dev_api_url: Url::from_str("http://127.0.0.1:8081")?,
+            faucet_url: None,
+        })
+    }
+
+    pub fn trove_testnet() -> Result<Self> {
+        Ok(Network {
+            name: String::from("trove"),
+            json_rpc_url: Url::from_str("http://trove.aws.hlw3truzy4ls.com")?,
+            dev_api_url: Url::from_str("https://api.trove.aws.hlw3truzy4ls.com")?,
+            faucet_url: Some(Url::from_str(
+                "https://trove.aws.hlw3truzy4ls.com/accounts",
```

Comment:
We touched on this before offline. `faucet_base_url` != `faucet_accounts_url`.

In the future, we'll be doing more than just account creation through the faucet, for example we'll be doing [funding of accounts](https://github.com/diem/diem/blob/main/client/faucet/src/lib.rs#L201-L209). If you only have `faucet_accounts_url`, it's clumsier to build `faucet_funding_url`, while if you have a `faucet_base_url`, we can construct both accounts and funding urls with relative paths.

TL;DR: Remove the trailing `/accounts` from the url.